lib/index.js

```diff
@@ -3,16 +3,32 @@ var nn = require('bindings')('node_nanomsg.node');
 var util = require('util');
 var EventEmitter = require('events').EventEmitter;
 
+/**
+ * generic socket-level NN_SOL_SOCKET options
+ */
+var sol = {
+  linger          : nn.NN_LINGER,
+  sndbuf          : nn.NN_SNDBUF,
+  rcvbuf          : nn.NN_RCVBUF,
+  sndtimeo        : nn.NN_SNDTIMEO,
+  rcvtimeo        : nn.NN_RCVTIMEO,
+  reconn          : nn.NN_RECONNECT_IVL,
+  maxreconn       : nn.NN_RECONNECT_IVL_MAX,
+  sndprio         : nn.NN_SNDPRIO,
+  rcvprio         : 9,
```

Comment:
I'm not sure how I feel about committing this in this PR. I think in a follow up, I'd like to have a test case that shows it's broken and a comment
```
// TODO: Issue #50
```